### ISSUE TITLE
Warning: Critical dependency: the request of a dependency is an expression

### ISSUE THREAD

**Issue by anisabboud** [firebase/firebase-js-sdk#6660]
Hi, I just updated Firebase SDK to the latest version released yesterday (9.11.0 - 2022-10-06), and now getting a console warning when building the project (with Angular ng build / ng serve):

![image](https://user-images.githubusercontent.com/5672108/194479939-c2153df5-0ee8-4102-aca7-b661131c6ef0.png)

```bash
./node_modules/@firebase/util/dist/index.esm2017.js:741:19-44 - Warning: Critical dependency: the request of a dependency is an expression
```

Specifically, the warning comes from this line:
![image](https://user-images.githubusercontent.com/5672108/194480409-32a62f96-2036-47ae-84ab-bfc742eabc00.png)

which seems to be in this file: **firebase-js-sdk/packages/util/src/deferred.ts**
https://github.com/firebase/firebase-js-sdk/blob/5ad7ff2ae955c297556223e6cb3ad9d4b897f664/packages/util/src/deferred.ts#L39

**Comment by huzaifa3165** [comment]
I'm also getting the same warning.

**Comment by hsubox76** [comment]
This is puzzling because that file hasn't been updated in 2 years. Did you upgrade directly from 9.10.0 to 9.11.0 or did you start from a version earlier than 9.10.0? Did you update any other dependencies as part of that update?

This is the first time `@firebase/util` has had a version bump since July so it's possible some commit to util between then and now caused it also.

Which products are you using (analytics, app-check, database, firestore, etc.)? It might also be that something changed in the way the product code calls `Deferred`.

**Comment by hsubox76** [comment]
Actually I looked up the error and it seems it's a webpack warning about using `require()` on a dynamic variable. In that case it's most likely caused by this line, which was added recently:

https://github.com/firebase/firebase-js-sdk/blob/0a112bd2aee2e709d6733b1a36876e6fae1e347f/packages/util/src/defaults.ts#L78

and it looks like the source map is a little bit off so it's pointing at a different location in the util bundle. I'll see what I can do to address this.

**Comment by nandorojo** [comment]
Seeing the same thing on Next.js. Thanks for looking into it @hsubox76.

**Comment by tomasconnor** [comment]
+1

**Comment by Amansg4** [comment]
Just had this problem. +1. Don't know a fix though

**Comment by nandorojo** [comment]
Downgrading to 9.10 fixed it for me.

**Comment by hsubox76** [comment]
The warning shouldn't affect any code functionality but you can downgrade to 9.10.0 if you want to avoid having a warning (or if I'm wrong and it is affecting your code). I have a fix (https://github.com/firebase/firebase-js-sdk/pull/6665) which should be released next week.

**Comment by dodomui** [comment]
+1

**Comment by meenal21** [comment]
+1. downgrading to 9.10.0 fixed the issue!

**Comment by BasilArackal** [comment]
> Hi, I just updated Firebase SDK to the latest version released yesterday (9.11.0 - 2022-10-06), and now getting a console warning when building the project (with Angular ng build / ng serve):
> 
> ![image](https://user-images.githubusercontent.com/5672108/194479939-c2153df5-0ee8-4102-aca7-b661131c6ef0.png)
> 
> ```shell
> ./node_modules/@firebase/util/dist/index.esm2017.js:741:19-44 - Warning: Critical dependency: the request of a dependency is an expression
> ```
> 
> Specifically, the warning comes from this line: ![image](https://user-images.githubusercontent.com/5672108/194480409-32a62f96-2036-47ae-84ab-bfc742eabc00.png)
> 
> which seems to be in this file: **firebase-js-sdk/packages/util/src/deferred.ts**
> 
> https://github.com/firebase/firebase-js-sdk/blob/5ad7ff2ae955c297556223e6cb3ad9d4b897f664/packages/util/src/deferred.ts#L39

I'm also getting the same warning on a Next.js project

**Comment by BasilArackal** [comment]
I'm also getting the same warning on a Next.js project

**Comment by sdk1990** [comment]
+1 on a Next.js project

**Comment by abwhb** [comment]
+1 on a Next.js project as well

**Comment by hsubox76** [comment]
As mentioned above, the fix has been merged and will be released this week.

**Comment by hsubox76** [comment]
The fix has been released in 9.12.0.
